synapse-core/src/main/java/com/tracelink/prodsec/synapse/scorecard/service/ScorecardService.java

```diff
@@ -130,23 +178,25 @@ public Scorecard getScorecardForFilter(String filterName) throws ProductsNotFoun
 	 * @throws ProductsNotFoundException if the project does not exist
 	 */
 	public Scorecard getScorecardForProject(String projectName) throws ProductsNotFoundException {
+		Scorecard scorecard = projectScorecardCache.get(projectName);
+		if (scorecard == null) {
+			throw new ProductsNotFoundException("Unknown Project Name");
```

Comment:
I think this might need to be changed a little now that you're looking at the cache. If someone adds a new project and then tries to retrieve the scorecard, the cache won't have a key for that project (within 15 minutes). Should this throw an exception (which I think is caught in ScorecardController), or return a blank scorecard? Before, we were throwing the exception if the project/product line/filter did not exist at all.